node/node.go

```diff
@@ -1413,46 +1448,36 @@ func (app *App) Start(ctx context.Context) error {
 	}
 
 	if err := app.startServices(ctx); err != nil {
-		return err
+		return nil, err
 	}
 
 	// need post verifying service to start first
 	app.preserveAfterRecovery(ctx)
 
 	if err := app.startAPIServices(ctx); err != nil {
-		return err
+		return nil, err
 	}
 
 	if err := app.launchStandalone(ctx); err != nil {
-		return err
+		return nil, err
 	}
 
 	events.SubscribeToLayers(app.clock)
 	app.log.Info("app started")
 
-	// notify anyone who might be listening that the app has finished starting.
-	// this can be used by, e.g., app tests.
-	close(app.started)
-
-	defer events.ReportError(events.NodeError{
-		Msg:   "node is shutting down",
-		Level: zapcore.InfoLevel,
-	})
-	// TODO: pass app.eg to components and wait for them collectively
-	if app.ptimesync != nil {
-		app.eg.Go(func() error {
-			app.errCh <- app.ptimesync.Wait()
-			return nil
-		})
-	}
-	// app blocks until it receives a signal to exit
-	// this signal may come from the node or from sig-abort (ctrl-c)
-	select {
-	case <-ctx.Done():
-		return nil
-	case err = <-app.errCh:
+	// cleanup shuts down components started here.
+	cleanup := func() error {
```

Comment:
I feel like this should be part of `app.stopServices` instead of being returned via an anonymous function.

Reply:
Yep, that looks a lot nicer thanks :D